sov-modules/sov-modules-api/src/bech32.rs

```diff
@@ -27,6 +27,21 @@ pub struct AddressBech32 {
     value: String,
 }
 
+impl AddressBech32 {
+    pub(crate) fn to_byte_array(&self) -> [u8; 32] {
+        let (_, data) = bech32_to_vec(&self.value).unwrap();
```

Comment:
(Non-blocking, out-of-scope): The name of this function made me expect that it would make a vec of the bech32 encoded data, not of the decoded data. Maybe we should rename it to `bech32_to_decoded_vec` or something?

Reply:
Renamed